ee/hogai/graph/root/nodes.py

```diff
@@ -653,7 +653,7 @@ async def arun(self, state: AssistantState, config: RunnableConfig) -> PartialAs
                         ui_payload={tool_call.name: result.artifact},
                         id=str(uuid4()),
                         tool_call_id=tool_call.id,
-                        visible=True,
+                        visible=state.show_tool_call_messages,
```

Comment:
We can use the field from the tool model:

```suggestion
                        visible=tool_call.show_tool_call_messages,
```

Reply:
Ah, true, this is much better